taskana-adapter-camunda-tasklistener/src/main/java/pro/taskana/adapter/camunda/tasklistener/TaskanaTaskListener.java

```diff
@@ -0,0 +1,114 @@
+package pro.taskana.adapter.camunda.tasklistener;
+
+import org.camunda.bpm.engine.delegate.DelegateTask;
+import org.camunda.bpm.engine.delegate.TaskListener;
+import org.camunda.bpm.engine.impl.context.Context;
+import org.camunda.bpm.model.bpmn.instance.camunda.CamundaProperties;
+import org.camunda.bpm.model.bpmn.instance.camunda.CamundaProperty;
+
+import java.sql.*;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.logging.Logger;
+
+public class TaskanaTaskListener implements TaskListener {
+
+    private final Logger LOGGER = Logger.getLogger(this.getClass().getName());
+
+    private static TaskanaTaskListener instance = null;
+
+    public static TaskanaTaskListener getInstance() {
+        if (instance == null) {
+            instance = new TaskanaTaskListener();
+        }
+        return instance;
+    }
+
+    public void notify(DelegateTask delegateTask) {
+
+        try {
+
+            Connection connection = Context.getProcessEngineConfiguration().getDataSource().getConnection();
+
+            try {
+                switch (delegateTask.getEventName()) {
+
+                    case "create":
+                        insertCreateEvent(delegateTask, connection);
+                        break;
+                    case "complete":
+                        insertCompleteEvent(delegateTask, connection);
+                        break;
+                    case "delete":
+                        insertDeleteEvent(delegateTask, connection);
+                        break;
+                }
+
+            } finally {
+                connection.close();
+            }
+
+        } catch (SQLException e) {
```

Comment:
All exceptions should be catched at this point. Otherwise the TaskanaTaskListener might render camunda non-functional.
Each exception should be logged using the logger and severity warning or error